sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/AccessOrderModel.java

```diff
@@ -0,0 +1,859 @@
+/*
+ * Copyright 2013-2023 Real Logic Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.co.real_logic.sbe.generation.java;
+
+import uk.co.real_logic.sbe.generation.Generators;
+import uk.co.real_logic.sbe.ir.Signal;
+import uk.co.real_logic.sbe.ir.Token;
+import org.agrona.collections.Int2ObjectHashMap;
+import org.agrona.collections.IntHashSet;
+import org.agrona.collections.IntObjConsumer;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.IntStream;
+
+import static uk.co.real_logic.sbe.ir.GenerationUtil.collectFields;
+import static uk.co.real_logic.sbe.ir.GenerationUtil.collectGroups;
+import static uk.co.real_logic.sbe.ir.GenerationUtil.collectVarData;
+
+// Lambdas without braces tend to conflict with the indentation Checkstyle expects.
+// Therefore, we allow lambdas with code blocks even when a lambda expression is possible.
+@SuppressWarnings("CodeBlock2Expr")
+final class AccessOrderModel
+{
+    private static final boolean GENERATE_ACCESS_ORDER_CHECKS = Boolean.parseBoolean(
+        System.getProperty("sbe.generate.access.order.checks", "true"));
```

Comment:
My personal preference would be to pull all of the property evaluation to the highest level possible (ideally the SbeTool, but practically would probably be TargetCodeGeneratorLoader) and then used the resolved values to configure the PrecedenceChecks.Context, rather than using the system properties as defaults via the Configuration class.  The benefit of this is two fold.  We can see all of the tool configuration in the same place and we prevent coupling of field types of the PrecedenceChecks.Context to types represent able as a system property.

Reply:
I'm happy to make that change.

> We can see all of the tool configuration in the same place

I'm unsure if it is by design, but the CSharp code generator is not included in `TargetCodeGeneratorLoader`.
Maybe it should be?
Without it there, the "decoding" of system properties might feel a bit scattered.